core/payment/src/api/allocations.rs

```diff
@@ -50,6 +52,42 @@ async fn create_allocation(
         .unwrap_or(DEFAULT_PAYMENT_PLATFORM.to_string());
     let address = allocation.address.clone().unwrap_or(node_id.to_string());
 
+    // If the request contains information about the payment platform, initialize the account
+    // by setting the `send` field to `true`, as it is implied by the intent behing allocation of funds.
+    if let Some(platform) = &allocation.payment_platform {
+        // payment_platform is of the form driver-network-token
+        // eg. erc20-rinkeby-tglm
+        let [driver, network, token]: [&str; 3] =
+            match platform.split("-").collect::<Vec<_>>().try_into() {
```

Comment:
I'm not sure, what will happen in case we will have for example 4 or 2 elements after split. Can you ensure it won't panic?

Reply:
It won't panic, an invalid number of elements will trigger the `Err(_e) => ...` match arm and return `response::bad_request(...)`.